### PR TITLE
Infoviews & Browser in QML

### PR DESCRIPTION
**Pull request by jherico**
This PR adds the infoview based dialogs to the 3D scene.  These are dialogs that display a webview with local HTML content, currently the 'about' dialog and the 'edit entities help' dialog.  

This PR adds a very simple browser window which can load web pages inside the offscreen UI

Known issue:  clicking in links in the dialogs results in the new content being opened in the current window rather than being launched in an external browser.  Most likely the external browser would defeat the purpose, but opening them in an 'internal browser' will have to wait on the creation of such a thing.

Known issue: currently on macs the WebEngine isn't being deployed due to https://bugreports.qt.io/browse/QTBUG-35211

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 14b660baf70b91d86f475e69e61e7f8dd28163c2 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8b814082e16ddc3dfd02ed5a70e8b7d223a2d64 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c799ecbf83eeda4d79a99ddbb846aa459a28c4d0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7bd8bbf5653b8354c0c3b0e6f9fc33532a6ab3b5 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7bd8bbf5653b8354c0c3b0e6f9fc33532a6ab3b5 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9836ffed97346a0b42c915e1d44bc9a4e1e7a84e available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0fb8884f9c16ebdab3e7f7c6977f18ef28c65d80 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4349e797ff258321847fe1d0cd7fd6101dbdab0 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Comment by murillodigital** [comment]
@jherico fixes to release packaging due to macdeployqt bug are in place, review/merge at will. Latest Mac PR build is functional.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fbee3c1782daec19d64208ca1064f8a7a51f88b5 available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

File: libraries/ui/src/InfoView.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+void InfoView::setUrl(const QUrl& url) {
+    if (url != _url) {
+        _url = url;
+        emit urlChanged();
+    }
+}
+
+
+//#include <QApplication>
```
Comment (ZappoMan) [inline libraries/ui/src/InfoView.cpp]: I'm ready to merge other than this issue of all this dead code here... we should probably just delete this code? Right?

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9de502e769a4921c248f2d921cdec346551fb4e available at: https://gist.github.com/ef973722a45915491957<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4701.dmg

**Pull request merged**